Sources/SocksCore/TCPSocket.swift

```diff
@@ -203,12 +203,12 @@ public class TCPInternetSocket: InternetSocket, TCPSocket, TCPReadableSocket, TC
     public func close() throws {
         if closed { return }
         stopWatching()
-        if socket_close(self.descriptor) != 0 {
+        closed = true
```

Comment:
Below we throw an error saying `closeFailed` so would setting it to `true` in the line above be dishonest about the current state of the socket?